pandas/core/nanops.py

```diff
@@ -909,10 +909,7 @@ def nanvar(values, *, axis=None, skipna=True, ddof=1, mask=None):
         if mask is not None:
             values[mask] = np.nan
 
-    if is_float_dtype(values.dtype):
```

Comment:
if it doesn't break anything else would take the change, pls add a test and a whatsnew note